openff/toolkit/topology/molecule.py

```diff
@@ -1163,6 +1170,27 @@ def to_dict(self) -> dict:
         molecule_dict
             A dictionary representation of the molecule.
 
+            - **name** (str): An optional name to be associated with the molecule
+            - **atoms** (list[dict]): A list of dictionary inputs for :meth:`Atom.from_dict()`
+            - **bonds** (list[dict]): A list of dictionary inputs for :meth:`Bond.from_dict()`
+            - **conformers** (list[list]): A list containing the cartesian coordinates of each atoms in the order
+              defined in ``atoms``.
+            - **properties** (dict): Outputs from chosen a chosen toolkit:
+
+                - **atom_map** (dict): Dictionary of atom index (as in ``atoms`` entry) and the mapped index relevant
+                  to a mapped canonical smiles string
+                - **\*\*kwargs**: Other toolkit dependent outputs
+
```

Comment:
This dict is empty by default, i.e. not including the atom map, so I'd prefer either not listing it here or being more general about what can go in. IIRC it's a dumping ground for just about anything that can be represented in memory

```python
In [10]: Molecule.from_smiles("CCO").to_dict()['properties']
Out[10]: {}

Reply:
In the doctoring of `to_smiles` it is encouraged for a user to supply an `atom_map` for this purpose, but doesn't define its format. I think it's important to define it, and I believe this is the place to do it.

Reply:
Ah totally my mistake, I didn't read the second half of that comment

IMHO
* I have a preference for whatever gets this over the finish line with the lowest friction (I think the original objective of structure of `Molecule`-generated dictionaries is more than met)
* The added docs/examples you're considering adding are valuable but may be better for different portions of the docs than the docstrings of some serialization methods. I know CMILES is covered in the molecule cookbook but I forget the details

Reply:
I see that it is "verbally" explained in `from_smiles` I can remove this if you want, or you can save me the commit.

Reply:
Either is fine with me, tiebreaker going to "do nothing because that's quicker"

(I tagged the unit string as the only blocker, everything else is up to you)